### PR TITLE
Add key bind to hotswap internal screen rotation

### PR DESCRIPTION
**Pull request by iota97**
As requested on Discord by Azureal:

Screenshot_2022-04-23_08-53-41

### CODE REVIEW

**Comment by hrydgard** [comment]
Hm. Maybe we should consider a single toggle key bind instead? Although four isn't too bad I suppose. Mainly concerned about the number of strings to translate, as usual...

**Comment by iota97** [comment]
A single hotkey is not enough to take care of both 90 and 270 degree rotation. I can do 2 if you prefer tho' :)

**Comment by hrydgard** [comment]
Why not, you could just have the same key toggle through the four directions :) And binding one key is easier than binding two (I think most people wouldn't use all four) plus you don't even have to think if you want portrait or reverse portrait for example, you just hit the key until it looks right.

Although, if four separate keybindings really is useful, I'm fine with it.

**Comment by unknownbrackets** [comment]
Hm.  Cycling through four sounds potentially annoying if it's when the game goes to a certain screen and you want to flip back and forth.

It'd be even more translations, but I suppose the other option would be a single bind to toggle orientation + a per game setting for what the alternate orientation is.  But personally I think all four is probably fine...

-[Unknown]

**Comment by hrydgard** [comment]
Hm yeah, I didn't consider the case of games that often switch rotations. 

Eh, let's go with all four.

File: Core/KeyMap.cpp [hunk 1]
```diff
@@ -395,6 +395,11 @@ const KeyMap_IntStrPair psp_button_names[] = {
 	{VIRTKEY_ANALOG_ROTATE_CW, "Rotate Analog (CW)"},
 	{VIRTKEY_ANALOG_ROTATE_CCW, "Rotate Analog (CCW)"},
 
+	{VIRTKEY_SCREEN_ROTATION_VERTICAL, "Portrait"},
+	{VIRTKEY_SCREEN_ROTATION_VERTICAL180, "Portrait reversed"},
+	{VIRTKEY_SCREEN_ROTATION_HORIZONTAL, "Landscape"},
+	{VIRTKEY_SCREEN_ROTATION_HORIZONTAL180, "Landscape reversed"},
```
Comment (hrydgard) [inline Core/KeyMap.cpp:401]: Will these be descriptive enough in the mapping UI? Although I guess "Screen Orientation Portrait Reverse" would be too long...

**Comment by unknownbrackets** [comment]
If I understood Discord, that was this case - I haven't played .hack//LINK, but sounded like it has a message center that expects you to rotate your PSP to read it.

-[Unknown]

File: Core/KeyMap.cpp (hunk 1 above)
Comment (unknownbrackets) [reply Core/KeyMap.cpp:401]: Maybe just "Display Portrait" or "Orient Portait" etc.?  Super long descriptions get awkward in that UI...

-[Unknown]

Comment (iota97) [reply Core/KeyMap.cpp]: Went for "Display ..." it wrap to new line in the reversed one, but it already happen for other language with other keys AFAIR.

**Review by hrydgard — APPROVED** [review]

**Pull request merged**